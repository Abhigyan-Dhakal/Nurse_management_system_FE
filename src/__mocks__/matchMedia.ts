export default Object.defineProperty(window, "matchMedia", {
  value: () => {
    return {
      matches: false,
      addListener: () => {},
      removeListener: () => {},
    };
  },
});
