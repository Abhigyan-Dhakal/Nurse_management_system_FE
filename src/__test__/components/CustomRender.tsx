import type { PreloadedState } from "@reduxjs/toolkit";
import { configureStore } from "@reduxjs/toolkit";
import type { RenderOptions } from "@testing-library/react";
import { render } from "@testing-library/react";
import React, { PropsWithChildren } from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { allReducers, RootState, store } from "../../redux/store";
export const setupStore = (preloadedState?: PreloadedState<RootState>) => {
  return configureStore({
    reducer: allReducers,
    preloadedState,
  });
};
interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<RootState>;
  store?: typeof store;
}
export function renderWithProvidersAndNavigation(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return (
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
