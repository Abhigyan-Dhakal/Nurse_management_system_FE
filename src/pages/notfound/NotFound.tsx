import "./NotFound.css";

type Props = {};

export const NotFound = (props: Props) => {
  return (
    <div className="notfound-container">
      <h1>ERROR 404! Page Not Found!</h1>
    </div>
  );
};
