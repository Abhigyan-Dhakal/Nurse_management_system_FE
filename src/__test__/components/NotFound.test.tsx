import "../../__mocks__/matchMedia";
import { NotFound } from "../../pages/notfound/NotFound";
import { renderWithProvidersAndNavigation } from "./CustomRender";

it("should render login screen properly", () => {
  const page = renderWithProvidersAndNavigation(<NotFound />);
  expect(page.asFragment()).toMatchSnapshot();
});
