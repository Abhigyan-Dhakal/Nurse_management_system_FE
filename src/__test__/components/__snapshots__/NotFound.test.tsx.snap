// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login screen properly 1`] = `
<DocumentFragment>
  <div
    class="notfound-container"
  >
    <h1>
      ERROR 404! Page Not Found!
    </h1>
  </div>
</DocumentFragment>
`;
